examples/new_design_examples/new_design_msv.m

```diff
@@ -0,0 +1,98 @@
+clear all
+close all
+import casadi.*
+import vdx.*
+
+R = 3.5;
+%% Define projected system
+problem_options = nosnoc.Options();
+solver_options = nosnoc.solver.Options();
+%problem_options.rk_scheme = RKSchemes.GAUSS_LEGENDRE;
+%problem_options.rk_scheme = RKSchemes.LOBATTO_IIIC;
+problem_options.rk_scheme = RKSchemes.RADAU_IIA;
+%problem_options.rk_representation= RKRepresentation.differential_lift_x; 
+problem_options.rk_representation = RKRepresentation.integral;
+problem_options.cross_comp_mode = CrossCompMode.STAGE_STAGE;
+problem_options.cross_comp_mode = CrossCompMode.FE_FE;
+problem_options.N_finite_elements = 2;
+problem_options.n_s = 2;
+problem_options.N_stages = 25;
+problem_options.T = 1;
+problem_options.rho_h = 1e-4;
+problem_options.time_optimal_problem = true;
+%solver_options.homotopy_steering_strategy = 'ELL_INF';
+solver_options.complementarity_tol = 1e-10;
+solver_options.print_level = 3;
+
+model = nosnoc.model.Pds();
+x1 = SX.sym('x1', 2);
+x2 = SX.sym('x2', 2);
+x = [x1;x2];
+x_target = [0;0;0;0];
+model.x = [x];
+model.lbx = [-inf;-inf;-inf;-inf];
+model.ubx = [inf;inf;inf;inf];
+x0 =[-25;-25;-15;-15];
+model.x0 = [x0];
+u1 = SX.sym('u1', 2);
+u2 = SX.sym('u2', 2);
+model.u = [u1;u2];
+model.lbu = [-100/sqrt(2);-100/sqrt(2);-60/sqrt(2);-60/sqrt(2)];
+model.ubu = [100/sqrt(2);100/sqrt(2);60/sqrt(2);60/sqrt(2)];
+%model.lbu = [-100/sqrt(2);-100/sqrt(2);0;0];
+%model.ubu = [100/sqrt(2);100/sqrt(2);0;0];
+model.u0 = model.ubu;
+model.c = [norm_2(x2-x1)-2*R];
+model.f_x = [u1;u2];
+
+% costs
+model.f_q = 0;
+model.f_q_T = 0.5*(x-x_target)'*(x-x_target);
+
+% Solve
+ocp_solver = nosnoc.ocp.Solver(model, problem_options, solver_options);
+ocp_solver.solve();
+
+%% plot
+fontsize = 12;
+x_res = ocp_solver.get('x');
+u_res = ocp_solver.get('u');
+u_rep = kron(u_res, ones(1,problem_options.N_finite_elements(1)));
+T_final = ocp_solver.get('T_final');
```

Comment:
if it is not a time optimal problem, this will throw an error.
maybe we just return then just the model.T in this case - so the code is not broken. 
often when solving time optimal problems, one tries a few fixed T solutions to debug

Reply:
I think we should return the value of `discrete_time_problem.p.T` (as the user may have modified it after the fact and it may not equal `model.T` but otherwise I agree and this is implemented in 'nosnoc.ocp.Solver.get'.